Yosemite/Yosemite/Stores/AccountCreationStore.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import Networking
+
+/// Handles `AccountCreationAction`
+///
+public final class AccountCreationStore: DeauthenticatedStore {
+    // Keeps a strong reference to remote to keep requests alive.
+    private let remote: AccountRemoteProtocol
+
+    private let dotcomClientID: String
+    private let dotcomClientSecret: String
+
+    public init(dotcomClientID: String, dotcomClientSecret: String, remote: AccountRemoteProtocol, dispatcher: Dispatcher) {
+        self.remote = remote
+        self.dotcomClientID = dotcomClientID
+        self.dotcomClientSecret = dotcomClientSecret
+        super.init(dispatcher: dispatcher)
+    }
+
+    public convenience init(dotcomClientID: String, dotcomClientSecret: String, network: Network, dispatcher: Dispatcher) {
+        let remote = AccountRemote(network: network)
+        self.init(dotcomClientID: dotcomClientID, dotcomClientSecret: dotcomClientSecret, remote: remote, dispatcher: dispatcher)
+    }
+
+    public override func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: AccountCreationAction.self)
+    }
+
+    /// Called whenever a given Action is dispatched.
+    ///
+    public override func onAction(_ action: Action) {
+        guard let action = action as? AccountCreationAction else {
+            assertionFailure("AccountCreationStore received an unsupported action: \(action)")
+            return
+        }
+        switch action {
+        case .createAccount(let email, let password, let completion):
+            createAccount(email: email, password: password, completion: completion)
+        }
+    }
+}
+
+private extension AccountCreationStore {
+    func createAccount(email: String, password: String, completion: @escaping (Result<CreateAccountResult, CreateAccountError>) -> Void) {
+        Task { @MainActor in
```

Comment:
❓ A dumb question, but the `@MainActor` here will guarantee that the `completion` will be triggered on the main thread, right? Will it also cause the network calls to be called on the main thread?

Reply:
I believe Alamofire executes the actual network request in a background thread, and the [response handling callback in the main thread by default](https://github.com/Alamofire/Alamofire/blob/master/Documentation/Usage.md#response-handler-queue). Other Yosemite stores also make calls to their Remote on the main thread (and actions have to be dispatched on the main thread).